### PR TITLE
create nexus-validation module, move over bits from common/core

### PR DESCRIPTION
**Pull request by jdillon**
related:

https://github.com/sonatype/nexus-pro/pull/520

### CODE REVIEW

**Comment by jdillon** [comment]
note: overly complex merge due to adjusting codestyle of imports to standard

**Comment by mcculls** [comment]
reminder to update: https://github.com/sonatype/nexus-oss/blob/master/components/nexus-extender/src/main/java/org/sonatype/nexus/extender/modules/NexusBundleModule.java#L150

**Comment by jdillon** [comment]
Adds custom @Case validator based on tutorial here https://docs.jboss.org/hibernate/validator/4.0.1/reference/en-US/html/validator-customconstraints.html for ref

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF179-1 :white_check_mark:

**Comment by cstamas** [comment]
+1

File: components/nexus-validation/src/main/java/org/sonatype/nexus/validation/constraint/CaseValidator.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * @since 3.0
+ */
+public class CaseValidator
+  implements ConstraintValidator<Case,String>
+{
+  private CaseType type;
+
+  @Override
+  public void initialize(final Case annotation) {
+    type = annotation.value();
```
Comment (mrprescott) [inline components/nexus-validation/src/main/java/org/sonatype/nexus/validation/constraint/CaseValidator.java]: Worth having checkNotNull here? I guess this is only called by framework code.

**Comment by mrprescott** [comment]
+1

File: components/nexus-validation/src/main/java/org/sonatype/nexus/validation/constraint/CaseValidator.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-validation/src/main/java/org/sonatype/nexus/validation/constraint/CaseValidator.java]: yes that is why

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**